### ISSUE TITLE
Error on starting the server

### ISSUE THREAD

**Issue by Abby1096** [satori-com/mzbench#149]
Hi

I pulled the commit '02be2684655cde94d537c322bb0611e258ae9718' and I'm using the Erlang OTP release 18.
Facing Issue on starting the server `./bin/mzbench start_server`

`===> Compiling mzbench_api`
`===> Compiling src/mzb_k8s_plugin.erl failed`
`src/mzb_k8s_plugin.erl:26: syntax error before: ':='`

`src/mzb_k8s_plugin.erl:34: type plugin_opts() undefined`

`make: *** [.make/compilation-up-to-date] Error 1`
`make: Leaving directory `/home/centos/mzbench/server'`

Can anyone help me with this !?

**Comment by parsifal-47** [comment]
Hi,
unfortunately, it looks like the release 18 is no longer supported, you may see the versions that it is currently checked against here: https://github.com/mzbench/mzbench/blob/master/.travis.yml

The place it breaks is a type-hint, it is not critical, but I'm not completely sure that the rest of the code is OK with erl18, you may try to remove lines 26-29 and see what happened if you can't upgrade to 19 or 20. Otherwise, consider upgrading

also please check this branch: https://github.com/mzbench/mzbench
it has some updates regarding k8 plugin

Thanks!

**Comment by Abby1096** [comment]
Now , I could able to run the server . I changed the syntax `:=` to `=>`
And it runs.
[Abby@ip-172-31-33-167 mzbench]$ ./bin/mzbench start_server
Executing make -C /home/centos/mzbench/bin/../server generate
Waiting for server application start
..Webserver is started at http://127.0.0.1:4800
No config file is used, to create one run:
cp ~/.config/mzbench/server.config.example ~/.config/mzbench/server.config
ok

But however I couldn't view it on browser .Anything I need to change ?

**Comment by parsifal-47** [comment]
is it running on the same host as your browser?
the reason I ask is: by default it is not opened for external connections, you need to change bind address from "127.0.0.1" to "0.0.0.0" if you need to connect from outside

**Comment by Abby1096** [comment]
Nope..I am running mzbench server in an instance..Then , Is it possible to view in my Local Browser ? Is there any configuration changes to allow the external connections in MzBench Server ?

**Comment by parsifal-47** [comment]
yes, you need to copy config file as it suggests:
```
cp ~/.config/mzbench/server.config.example ~/.config/mzbench/server.config
```
and then change "127.0.0.1" to "0.0.0.0" in it, please let me know if it works

**Comment by parsifal-47** [comment]
more precisely, it should be
`{network_interface, "127.0.0.1"},`

and you need to change it to `0.0.0.0`

**Comment by parsifal-47** [comment]
`enoent` could indicate that server has no privileges to open the config file, please check access rights

**Comment by Abby1096** [comment]
[centos@ip-172-31-32-208 ~]$ ll ~/.config/mzbench/
-rw-rw-r--. 1 centos centos 1334 Dec  7 05:13 server.config
-rw-rw-r--. 1 centos centos 1336 Dec  7 05:03 server.config.example

**Comment by timofey-barmin** [comment]
It complains about "/home/centos/mzbench//.config/mzbench/server.config" but your config is in ~/.config/mzbench/
I guess you need to move it to /home/centos/mzbench//.config/mzbench/ :)

**Comment by Abby1096** [comment]
`/home/centos/mzbench//.config/mzbench/`
I couldn't Get this path in my folder...

**Comment by parsifal-47** [comment]
interesting, it could be a bug while resolving "~", just to make sure I understand you:
there is no path `/home/centos/mzbench//.config/mzbench/` in your system, so it tries to find config there and fails, what if you create it?
I understand that generally it is maybe not a good idea, but just to make sure we diagnose it correctly
